text/0080-datetime-extension.md

```diff
@@ -0,0 +1,367 @@
+- `.toHours()` returns a `long` describing the number of hours in this duration. (`.toMinutes() / 60`)
+- `.toDays()` returns a `long` describing the number of days in this duration. (`.toHours() / 24`)
+
+Values with type `duration` duration can also be used with comparison operators:
+
+- `DT1 < DT2` returns `true` when `DT1` is before `DT2`
+- `DT1 <= DT2` returns `true` when `DT1` is before or equal to `DT2`
+- `DT1 > DT2` returns `true` when `DT1` is after `DT2`
+- `DT1 >= DT2` returns `true` when `DT1` is after or equal to `DT2`
+- `DT1 == DT2` returns `true` when `DT1` is equal to `DT2`
+
+### Out of scope
+
+- **Conversion between UTC and epochs:** This will be particularly difficult to model and verify in Lean (although it's technically possible, see [this paper](https://dl.acm.org/doi/abs/10.1145/3636501.3636958) which does something similar in the Coq proof assistant). Since it will likely require input-dependent loops, it is unlikely that this can be reasoned about efficiently with SMT.
+
+- **Conversion between UTC and other Time Zones:** Time Zones are a notoriously complex system that evolves rapidly. We avoid this complexity by offering `datetime.add(duration).` Policy authors that require "local time" can either provide an additional datetime in `context` or provide a `duration` to the `context` and call `.add()` to shift the time.
+
+- **Leap Seconds:** Cedar does not have a clock, and this proposal does not add one. Instead, we pass Unix Time through `context` or an entity and let the system / application handle leap seconds.
+
+- **Utilities to get date components**:  It is difficult to model `isLeapYear` in SMT, which makes it hard to support `datetime.dayOfWeek()`, `datetime.dayOfMonth()` or `datetime.year()`. Cedar applications that wish to define policy based on these ideas should pass pre-computed properties through entities or through the `context`.
+
+### Support for date/time in other authorization systems
+
+AWS IAM supports [date condition operators](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html#Conditions_Date), which can check relationships between date/time values in the ISO 8601 date format or Unix time. You can find an example of a IAM policy using date/time [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws-dates.html).
+
+[Open Policy Agent](https://www.openpolicyagent.org) provides a [Time API](https://www.openpolicyagent.org/docs/latest/policy-reference/#time) with nanosecond precision and extensive time zone support. During policy evaluation, the current timestamp can be returned, and date/time arithmetic can be performed. The `diff` (equivalent to our proposed `since` function) returns an array of positional time unit components, instead of a value typed similarly to our proposed `duration`.
+
+## Alternatives
+
```

Comment:
I wouldn't call these alternatives: they are workarounds. That's because they don't require changes to Cedar, they just present ways to use it as it exists now.

Reply:
Correct. But they are alternatives because we can not implement the proposal in this RFC and advocate for one of these alternatives instead... accepting the tradeoffs of doing so.

Reply:
Fair enough. In the past we've specifically highlighted a way you could do without the proposal as a "workaround", just to help the reader a little. That's different than an Alternative implementation/change. See for example this [workaround](https://github.com/cedar-policy/rfcs/blob/main/text/0068-entity-tags.md#workaround-records-with-optional-attributes) for entity attribute maps. Maybe just add the "workaround" prefix to these parts?